CSS layout cookbook: contribute page (#29993)
mdn/content: files/en-us/web/css/layout_cookbook/contribute_a_recipe/index.md

**The simplest possible version of any useful web design pattern**. Each line of CSS that you include should be there to help demonstrate the pattern, so leave out anything there purely to make the example look pretty. The idea is that someone can take the pattern and use it in a site with their own styles.

A recipe consists of:

- A live example, stored in the [CSS Examples GitHub repo](https://github.com/mdn/css-examples).
- A downloadable version of that example, also stored in the CSS Examples repo.
- A page in the [CSS layout cookbook](/en-US/docs/Web/CSS/Layout_cookbook) section of the site, which should include the following components:

  1. Introduction
  2. Requirements
  3. Recipe
  4. Decisions
  5. Useful fallbacks or alternative methods (when relevant)
  6. Accessibility concerns
  7. Browser compatibility (only for CSS properties not fully supported)
  8. Additional resources

## Steps to publish a recipe

To create a recipe and add it to the CSS Layout Cookbook, follow these steps:

> 1. [Build a pattern](#1-build-a-pattern)
> 2. [Create a live example](#2-create-a-live-example)
> 3. [Create a downloadable version](#3-create-a-downloadable-version)
> 4. [Open a pull request](#4-open-a-pull-request-with-your-example)
> 5. [Add your content to MDN](#5-create-your-page)

Reviewer: Checked these links in the local preview. The hyphens need to be underscores. Plus there is a period after the numbers in the headings

```suggestion
1. [Build a pattern](#1._build_a_pattern)
2. [Create a live example](#2._create_a_live_example)
3. [Create a downloadable version](#3._create_a_downloadable_version)
4. [Open a pull request](#4._open_a_pull_request_with_your_example)
5. [Add your content to MDN](#5._create_your_page)
```

#### Useful tips

1. Do not indent the CSS and HTML inside the textareas; instead, bump the code up against the start of the lines. This will look better when it is rendered.
2. If any images are required, put them into the directory with the examples. You are welcome to use any of the images already there.
3. You can adjust the height of the textareas by changing the height in the inline styles.

### 3. Create a downloadable version

As the examples include all of our live example code, we want to offer visitors the code for their example without anything they don't need. Make a copy of your live example with `--download` appended. For example, `center.html`, has a download version named `center--download.html`. This file will be similar to your initial example. It should be a basic version of your pattern as a single HTML page.

It can be handy to include the body CSS rules that are in our included stylesheet. Take a look at [`cookbook-template--download.html`](https://github.com/mdn/css-examples/blob/main/css-cookbook/cookbook-template--download.html); these are included as a starting point.

### 4. Make a Pull Request with your example

> Make a [PR](/en-US/docs/MDN/Community/Pull_requests#open_a_pull_request) on the [CSS Examples repo](https://github.com/mdn/css-examples). This way, we can help you with any changes to the example that may be needed before you create your page. Plus, the example will need to be live to be included. This is why it makes sense to create the code examples first, before the explanatory guide. In your PR, explain the pattern and what you are demonstrating.

Reviewer: Shouldn't the PR link here point to the PRs in that repo? If we want, we can add a separate sentence about general guidelines for opening pull requests
```suggestion
Open a [pull request (PR)](https://github.com/mdn/css-examples/pulls) on the [CSS Examples repo](https://github.com/mdn/css-examples). This allows us to help you with any changes to the example that may be needed before you create your page. Plus, the example will need to be live to be included. This is why it makes sense to first create the code examples, and then the explanatory guide. In your PR, explain the pattern and what you are demonstrating.
```

> 
> Open a [pull request (PR)](/en-US/docs/MDN/Community/Pull_requests#open_a_pull_request) on the [CSS Examples repo](https://github.com/mdn/css-examples/pulls). This allows us to help you with any changes to the example that may be needed before you create your page. Plus, the example will need to be live to be included. This is why it makes sense first to create the code examples, and then the explanatory guide. In your PR, explain the pattern and what you are demonstrating.

Reviewer: - Removed extra line
- Updated this sentence "This is why it makes sense to first create the code examples and then the explanatory guide."
- Moving the PR link to the end
```suggestion
Open a pull request (PR) on the [CSS Examples repo](https://github.com/mdn/css-examples/pulls). This allows us to help you with any changes to the example that may be needed before you create your page. Plus, the example will need to be live to be included. This is why it makes sense to first create the code examples and then the explanatory guide. In your PR, explain the pattern and what you are demonstrating. See our [guidelines for opening a pull request](/en-US/docs/MDN/Community/Pull_requests#open_a_pull_request).
```

### 5. Create your page

> Once your example is merged, you can [open a pull request](/en-US/docs/MDN/Community/Pull_requests#open_a_pull_request) to create a new page in the [Layout Cookbook directory](https://github.com/mdn/content/tree/main/files/en-us/web/css/layout_cookbook).

Reviewer: ```suggestion
After your example PR is merged, open a pull request to create a new page in the [Layout Cookbook directory](https://github.com/mdn/content/tree/main/files/en-us/web/css/layout_cookbook). See our [guidelines for opening a pull request](/en-US/docs/MDN/Community/Pull_requests#open_a_pull_request).
```

There is [a template for Cookbook pages](https://github.com/mdn/content/blob/main/files/en-us/web/css/layout_cookbook/contribute_a_recipe/cookbook_template/index.md?plain=1) that you should use.
The template explains what should be part of each section, and you can refer to other cookbook examples for more help.

> Also, add a link to the new recipe in the top-level [Layout Cookbook page](/en-us/web/css/layout_cookbook/).

Reviewer 2: The bot is reporting a link flaw because of the trailing slash.

> Also, add a link to the new recipe in the top-level [Layout Cookbook page](/en-us/web/css/layout_cookbook/).

Author: ```suggestion
Also, add a link to the new recipe in the top-level [Layout Cookbook page](/en-us/web/css/layout_cookbook).
```

> After your example PR is merged, open a pull request to create a new page in the [Layout Cookbook directory](https://github.com/mdn/content/tree/main/files/en-us/web/css/layout_cookbook). See our [guidelines for opening a pull request](/en-US/docs/MDN/Community/Pull_requests#open_a_pull_request).

Reviewer: Moving PR guideline link to the previous step
```suggestion
After your example PR is merged, open a pull request to create a new page in the [Layout Cookbook directory](https://github.com/mdn/content/tree/main/files/en-us/web/css/layout_cookbook).
```

> If you have any questions or want someone to take a look at your page, get in touch via any [communication channel](/en-US/docs/MDN/Community/Communication_channels).

Reviewer: ```suggestion
If you have any questions or want someone to take a look at your page, get in touch via any of our [communication channels](/en-US/docs/MDN/Community/Communication_channels).
```
